### PR TITLE
Clarify parts of the "Configuration" documentation.

### LINKED ISSUE #34: docs: PEP 621 guide
I had some feedback for the PEP 621 guide:

* `version` doesn't need to be specified or be static in PEP 621; tools often have SCM or file based methods to get it. I think you are thinking about the METADATA record field, which does need it statically defined. I'd clarify that this is currently whey specific.
* One of the lines under version should be under name instead (Ideally, the name should be normalized to lowercase, with underscores replaced by hyphens).
* The inline table form might be nicer for the readme and license? I'd also show the most common readme example, `readme = { file="README.md", content-type="text/markdown"}`
* I'd point out that the description is the "short" description, or that it's supposed to be short. `readme` is the long one.
* I think the author example should have at least one with name and email, as that's the most common, I'd expect. Probably the inline form is best, as TOML out-of-line lists are ugly, IMO.

I'd also include a full-sized example at the end or beginning, showing most of the options in a reasonable project.

Nice that there's a tool section too :)

### PR DESCRIPTION
**Pull request by domdfcoding**
Closes #34

### CODE REVIEW

**Comment by domdfcoding** [comment]
@henryiii would you be able to take a look at this? 
I think I addressed all of your comments.

File: doc-source/configuration.rst [hunk 1]
```diff
@@ -110,10 +121,16 @@ As a minimum, the table MUST contain the keys ``name`` and ``version``.
...
-		content-type = "text/x-rst"
-		encoding = "UTF-8"
+	.. code-block:: TOML
+
+		[project]
+		readme = {
+			file = "README.md" ,
+			content-type = "text/markdown",
+			encoding = "UTF-8",
+		}
```
Comment (henryiii) [inline doc-source/configuration.rst:131]: This is not valid TOML syntax. Inline tables _must_ be a single line, and must not have a trailing comma. I find it _really_ irritating - "users", not "authors" write TOML and get tripped up by this. It's allowed on arrays, but not tables.

**Review by henryiii — APPROVED** [review]
I think this looks like a solid improvement, thanks!

**Comment by Jcreate-source** [comment]
I need to print on my comments  to trade add a trading bot on my api

**Comment by domdfcoding** [comment]
Thanks, I'll merge this then.

Do let me know if you have any more suggestions.

**Pull request merged**